### PR TITLE
Nodejs/week2

### PR DESCRIPTION
**Pull request by zmzade**
For the last part of exercise "What if the users writes a query parameter that is not supported?" could not find solution.

### CODE REVIEW

File: nodejs/week2/homework/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+app.get("/", (req, res) => res.send("nodejs week2 homework"));
+app.get("/numbers/add", (req, res) => {
+  res.send(`${parseInt(req.query.first) + parseInt(req.query.second)}`);
+});
+
+app.get("/numbers/multiply/:firstNum/:secondNum", (req, res) => {
+  res.send(`${parseInt(req.params.firstNum) * parseInt(req.params.secondNum)}`);
+});
+
```
Comment (laila1222) [inline nodejs/week2/homework/warmup/index.js:13]: Well done!

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  }*/
+
+  return response.status(200).json(mealsWithId[0]);
+  //console.log("in /api/meals");
+});
+
+//--------------------------------------------------
+router.get("/", async (request, response) => {
+  //What if the users writes a query parameter that is not supported?
+  ////////////////////////////////////
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:26]: Good point. In general the server must validate if the incoming parameters (in query, path or body) are valid. For this you can create a validation function. For example you can validate if the maxPrice is a number. If it is not, we should throw an error (return an error status). 
Similarly, if the query parameter is not maxPrice, limit, createdAfter, etc, you can also throw an error (return an error status).

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    if (request.query.limit > meals.length) {
+      return response.status(400).send("Not Found: Sorry! Less than this");
+    }
+    return response.status(200).send(limitedMeals);
+  }
+  //--------
+  //Respond with the json for all the meals
+
+  return response.status(200).json(meals);
+});
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:81]: Great job!

File: nodejs/week2/homework/meal-sharing/src/backend/api/reserv-router.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+});
+
+router.get("/", async (request, response) => {
+  try {
+    response.json(reservations);
+    console.log("in /api/meals");
+  } catch (error) {
+    throw error;
+  }
+});
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reserv-router.js:25]: Well done

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+router.get("/", async (request, response) => {
+  try {
+    response.json(reviews);
+    console.log("in /api/meals");
+  } catch (error) {
+    throw error;
+  }
+});
+
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js:26]: Good!

**Review by laila1222 — APPROVED** [review]
Excellent job!! 👍 🥇  
I saw that you had a question in a comment -> I gave a short answer to that.

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js (hunk 5 above)
Comment (zmzade) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js:26]: So thanks for explanation

**Pull request merged**